src/bufr/BufrParser/BufrMnemonicSet.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * (C) Copyright 2020 NOAA/NWS/NCEP/EMC
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#include <ostream>
+#include <vector>
+
+#include "BufrMnemonicSet.h"
+#include "BufrTypes.h"
+
+namespace Ingester
+{
+    BufrMnemonicSet::BufrMnemonicSet(const std::vector<std::string>& mnemonics,
+                                     const Channels& channels) :
+        mnemonics_(mnemonics),
+        mnemonicsStr_(makeMnemonicsStr(mnemonics)),
+        channels_(channels),
+        maxColumn_(findMaxChannel(channels))
+    {
+    }
+
+    std::string BufrMnemonicSet::makeMnemonicsStr(std::vector<std::string> mnemonics)
+    {
+        std::ostringstream mnemonicsStrStream;
+        for (auto mnemonicsIt = mnemonics.begin();
+             mnemonicsIt < mnemonics.end();
+             mnemonicsIt++)
+        {
+            mnemonicsStrStream << *mnemonicsIt;
+
+            if (mnemonicsIt != mnemonics.end() - 1)
+            {
+                mnemonicsStrStream << " ";
+            }
+        }
+
+        return mnemonicsStrStream.str();
+    }
+
+    size_t BufrMnemonicSet::findMaxChannel(const Channels& channels)
```

Comment:
If Channels is a ```std::set<int>```, then the max channel number is always an int. So the return type should be an int and there should be no need for the static_cast in the function.

Reply:
Or, change Channels to a set of size_t.